src/js/jsx/shared/TextInput.jsx

```diff
@@ -272,16 +272,19 @@ define(function (require, exports, module) {
             }
             selectDisabled = selectDisabled || false;
 
-            var node = React.findDOMNode(this.refs.input);
-            node.removeAttribute("readOnly");
-            node.removeAttribute("disabled");
-            node.focus();
-
-            this.setState({
-                editing: true,
-                selectDisabled: selectDisabled
-            });
-            this.acquireFocus();
+            this.acquireFocus()
+                .bind(this)
+                .then(function () {
+                    this.setState({
+                        editing: true,
+                        selectDisabled: selectDisabled
+                    }, function () {
+                        var node = React.findDOMNode(this.refs.input);
```

Comment:
I don't understand what's going on in this failure case. First, it seems like this should never fail, since in `acquireFocus` you're explicitly catching all errors. (I also don't think it makes sense in hindsight to catch errors from `acquireFocus`.) Second, I guess I just don't understand why you're doing what you're doing in the case of a failure.